src/main/webapp/scripts/map.js

```diff
@@ -13,98 +13,73 @@
 // limitations under the License.
 
 /**
- * Map declaration and initial setup
- * At the beginning, retrieves the API_KEY from an external file
+ * Function that queries salecards from servlet.
  */
-$(function () {
-  jQuery.get('../../../../API_KEY.txt', function (textString) {
-    const API_KEY = textString;
+async function querySalecard() {
+  await updateLocation();
+
+  // Object to draw the map.
+  const clientInfo = {
+    latitude: parseFloat(document.getElementById('lat').value),
+    longitude: parseFloat(document.getElementById('lng').value),
+    radius: Number(document.getElementById('distance').value),
+    markerName: 'Your Position',
+    type: 'client',
+    drawCircleRadius: true
+  }
 
-    // Create the script tag, set the appropriate attributes.
-    const scriptMapTag = document.createElement('script');
-    scriptMapTag.src = `https://maps.googleapis.com/maps/api/js?key=${API_KEY}`;
-    scriptMapTag.defer = true;
+  // Draw client information.
+  const map = drawMap(clientInfo);
 
-    document.head.appendChild(scriptMapTag);
-  });
-});
+  // Add vendor's markers.
+  fetchVendors(map);
+}
 
 /**
- * Function that queries salecards from servlet.
- * TODO: implement fetch from servlet
+ * Function that adds vendor marks to the map
+ * @param {Object} map
  */
-querySalecards = () => {
-
-  const map = new google.maps.Map(
-    document.getElementById('map'), { zoom: 15 });
-
-  // Get client current location.
-  if (navigator.geolocation) {
-    navigator.geolocation.getCurrentPosition((position) => {
-      const userPosition = {
-        lat: position.coords.latitude,
-        lng: position.coords.longitude
-      };
-      map.setCenter(userPosition);
-      map.setOptions({ styles: MAP_THEME });
-
-      // The marker, positioned at client's location.
-      const marker = new google.maps.Marker({
-        icon: {
-          url: 'http://maps.google.com/mapfiles/ms/icons/blue-dot.png'
-        },
-        map: map,
-        position: userPosition,
-        title: 'Your position'
-      });
-
-      // Declare circle that match user's query parameters.
-      const userCircle = new google.maps.Circle({
-        center: userPosition,
-        fillColor: '#FF0000',
-        fillOpacity: 0.20,
-        map,
-        radius: parseInt(document.getElementById('distance').value),
-        strokeColor: '#FF0000',
-        strokeOpacity: 0.8,
-        strokeWeight: 2
-      });
-      // Add vendor's markers.
-      addVendorsToMap(map);
-    },
-      (error) => {
-        alert(error.message);
+const fetchVendors = (map) => {
+  const params = getQueryParams();
+
+  fetch('/get-nearby-vendors', {
+    method: 'POST',
+    body: params
+  })
+      .then(response => response.json())
```

Comment:
Also here should be a good idea to handle the response by checking the status first since this servlet can also send errors too.